test/auth_cluster/test_auth_v2_migration.py

```diff
@@ -0,0 +1,141 @@
+            "rows": [
+                ("users", "user 1"),
+                ("users", "user 2"),
+            ]
+        },
+        {
+            "query": "INSERT INTO system_auth.role_attributes (role, name, value) VALUES (?, ?, ?)",
+            "rows": [
+                ("users", "service_level", "sl:fefe"),
+            ]
+        },
+    ]
+
+async def populate_test_data(manager: ManagerClient, data):
+    cql = manager.get_cql()
+    for d in data:
+        stmt = cql.prepare(d["query"])
+        stmt.consistency_level = ConsistencyLevel.ALL
+        await asyncio.gather(*(
+            cql.run_async(stmt.bind(row_data)) for row_data in d["rows"]))
+
+async def populate_auth_v1_data(manager: ManagerClient):
+    await populate_test_data(manager, auth_data())
+    # test also absence of deleted data
+    await populate_test_data(manager, [
+        {
+            "query": "INSERT INTO system_auth.roles (role, can_login, is_superuser, member_of, salted_hash) VALUES (?, ?, ?, ?, ?)",
+            "rows": [
+                ("deleted_user", True, False, None, "fefe"),
+            ]
+        },
+        {
+            "query": "DELETE FROM system_auth.roles WHERE role = ?",
+            "rows": [
+                ("deleted_user",),
+            ]
+        },
+    ])
+
+async def check_auth_v2_data_migration(manager: ManagerClient, hosts):
+    cql = manager.get_cql()
+    # auth reads are eventually consistent so we need to make sure hosts are up-to-date
+    assert hosts
+    await asyncio.gather(*(read_barrier(cql, host) for host in hosts))
+
+    data = auth_data()
+
+    roles = set()
+    for row in await cql.run_async("SELECT * FROM system_auth_v2.roles"):
+        member_of = frozenset(row.member_of) if row.member_of else None
+        roles.add((row.role, row.can_login, row.is_superuser, member_of, row.salted_hash))
+    assert roles == set(data[0]["rows"])
+
+    role_members = set()
+    for row in await cql.run_async("SELECT * FROM system_auth_v2.role_members"):
+        role_members.add((row.role, row.member))
+    assert role_members == set(data[1]["rows"])
+
+    role_attributes = set()
+    for row in await cql.run_async("SELECT * FROM system_auth_v2.role_attributes"):
+        role_attributes.add((row.role, row.name, row.value))
+    assert role_attributes == set(data[2]["rows"])
+
+async def check_auth_v2_works(manager: ManagerClient, hosts):
+    cql = manager.get_cql()
+    roles = set()
+    for row in await cql.run_async("LIST ROLES"):
+        roles.add(row.role)
+    assert roles == set(["cassandra", "user 1", "user 2", "users"])
+
+    user1_roles = await cql.run_async("LIST ROLES OF 'user 1'")
+    assert len(user1_roles) == 2
+    assert set([user1_roles[0].role, user1_roles[1].role]) == set(["users",  "user 1"])
+
+    await cql.run_async("CREATE ROLE IF NOT EXISTS user_after_migration")
+    await cql.run_async("DROP ROLE user_after_migration")
```

Comment:
Let's add a check at the end that this does operate on the correct tables (new ones, not the old ones) i.e. all modifications are resembled in those new tables as expected, and nothing changed in the old tables.

Reply:
ok